Add program-data separation to pybindings

lucylq opened the pull request:
Summary: Add support for optional data path for pybindings.

Differential Revision: D76353209

JacobSzwejbka commented on extension/pybindings/test/test_pybindings.py:
```diff
@@ -600,3 +601,32 @@ def test_method_method_meta(self) -> None:
+                tensor_data = bytes(exec_program._tensor_data.pop("_default_external_constant"))
+                ptd.write(tensor_data)
+            
+            executorch_program = self.runtime._load_for_executorch(pte_file, ptd_file)
```
probably need to update the pyi

mergennachin reviewed (commented):
https://github.com/pytorch/executorch/blob/cac1a71a7f4224f074150abe25ab91340fb037f4/runtime/__init__.py#L199-L204

lucylq committed:
Add program-data separation to pybindings (#13886)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/13886

Pull request resolved: https://github.com/pytorch/executorch/pull/13886

Add support for optional data path for pybindings.

Reviewed By: JacobSzwejbka

Differential Revision: D76353209
